### ISSUE TITLE
stylefunction.js?4bb2:118 Uncaught TypeError: text.match is not a function

### ISSUE THREAD

**Issue by uct8086** [openlayers/ol-mapbox-style#235]
i saw the issue #87 ,and i get the same error with wepack 4.0 , babel 6 on "ol-mapbox-style": "^4.3.1",
this is my code: 

```

      import olms from 'ol-mapbox-style';

      this.map = new Map({
            target: this.id,
            view
        });
        await olms(this.map, styleJson); // add Mapbox Style 
        this.map.addLayer(this.vector);
```

**Comment by uct8086** [comment]
by the way , the last version is ok, with webpack 2.0, babel 6

**Comment by ahocevar** [comment]
Webpack 2 sound strange. You should be able to get things running with Webpack 4. Make sure you are using ol-mapbox-style@5.0.0-beta.3.

**Comment by uct8086** [comment]
upgrade to ol-mapbox-style@5.0.0-beta.3 got this :

```
stylefunction.js?4bb2:633 Uncaught TypeError: text.setRotateWithView is not a function
    at styleFunction (stylefunction.js?4bb2:633)
    at CanvasVectorTileLayerRenderer.render (VectorTileLayer.js?b7fa:201)
    at loop (VectorTileLayer.js?b7fa:226)
    at CanvasVectorTileLayerRenderer.createReplayGroup_ (VectorTileLayer.js?b7fa:233)
    at CanvasVectorTileLayerRenderer.getTile (VectorTileLayer.js?b7fa:117)
    at CanvasVectorTileLayerRenderer.prepareFrame (TileLayer.js?b9ec:186)
    at CanvasVectorTileLayerRenderer.prepareFrame (VectorTileLayer.js?b7fa:143)
    at CanvasMapRenderer.renderFrame (Map.js?8eca:157)
    at Map.renderFrame_ (PluggableMap.js?fe37:1226)
    at Map.eval (PluggableMap.js?fe37:189)

```

**Comment by uct8086** [comment]
I fixed this bug , my bro .
on version 4.3.1 `"ol-mapbox-style": "^4.3.1",`, file stylefunction.js at line 374 :
```
icon = fromTemplate(iconImage, properties);
```
when the iconImage is Object type , it will throw an error like the issue describe, so i fixed it like this : 

```
function fromTemplate(text, properties) {
  if(typeof text == 'object') return;
  let parts;
  do {
    parts = text.match(templateRegEx);
    if (parts) {
      const value = properties[parts[2]] || '';
      text = parts[1] + value + parts[3];
    }
  } while (parts);
  return text;
}
```
now it running good.

**Comment by uct8086** [comment]
> That issue was also fixed with #233 and is in the latest release (v5.0.2).

great !
